src/ngraph_backend_manager.h

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+// The backend manager class is a singelton class that interfaces with the
+// bridge to provide necessary backend
+
+#ifndef NGRAPH_TF_BRIDGE_BACKEND_MANAGER_H_
+#define NGRAPH_TF_BRIDGE_BACKEND_MANAGER_H_
+
+#include <mutex>
+#include <ostream>
+#include <vector>
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/runtime/backend_manager.hpp"
+#include "ngraph_log.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+struct Backend {
+  unique_ptr<ng::runtime::Backend> backend_ptr;
+  mutex backend_mutex;
+};
+
+class BackendManager {
+ public:
+  // Returns the backend name currently set
+  static string GetCurrentlySetBackendName() {
+    NGRAPH_VLOG(5) << "Getting backend";
+    return BackendManager::ng_backend_name_;
+  };
+
+  // Returns the nGraph supported backend names
+  static unordered_set<string> GetSupportedBackendNames();
+
+  static bool IsSupportedBackend(string& backend_name);
+
+  static Status SetBackendName(const string& backend_name);
+
+  static void CreateBackendIfDoesNotExist(const string& backend_name);
+
+  // Returns a backend pointer of the type specified by the backend name
+  static ng::runtime::Backend* GetBackend(const string& backend_name);
+
+  // LockBackend
+  static void LockBackend(const string& backend_name);
+
+  // UnlockBackend
+  static void UnlockBackend(const string& backend_name);
+
+ private:
+  static string ng_backend_name_;  // currently set backend name
```

Comment:
`GUARDED_BY(ng_backend_name_mutex_)`

Reply:
Yes, was trying to add just that getting compilation errors